pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -91,6 +91,42 @@ type PingConfig struct {
 	Timeout time.Duration
 }
 
+// RPCConfig configures RPC serving.
+type RPCConfig struct {
+	// Network addresses to listen Neo RPC on. Each element must be a valid TCP
+	// address in 'host:port' format.
+	//
+	// Optional: by default, Neo RPC is not served.
+	Addresses []string
+
+	// Additional addresses that use TLS.
+	//
+	// Optional.
+	TLSConfig
+}
+
+// TLSConfig configures additional RPC serving over TLS.
+type TLSConfig struct {
+	// Additional TLS serving switcher.
+	//
+	// Optional: by default TLS is switched off.
+	Enabled bool
+
+	// Network addresses to listen Neo RPC on if Enabled. Each element must be a valid TCP
+	// address in 'host:port' format.
+	Addresses []string
+
+	// TLS certificate file path.
+	//
+	// Required if Enabled and more than one address is provided.
```

Comment:
what if single address is provided? I don't see this statement in `config/example/ir.yaml`

if it's required then `New` must assert this

Reply:
It should be required even if only one address is provided, isn't it?

Reply:
it is bad wording in fact; that is not smth that i meant, fixed

> if it's required then New must assert this

well, do not agree with you totally cause then it even can go deeper (check if the file exists, check if it _is a key/cert_, etc) but fixed

Reply:
yes, check all this stuff. Docs are written not to be nice but have strict behavior

Reply:
i meant what if it _is set_ and its value is `""`? what if `" "`? `blockchain` either can use provided config, or returns `error` from its methods, why perform checks that will be performed anyway?

Reply:
reasonable questions, I expect `New` to fail. Could you pls try this?

Reply:
if we speak about our `New`, it fails [now](https://github.com/nspcc-dev/neofs-node/pull/2345#discussion_r1197936419). if `neo-go` is our goal, than it does not touch cert and key until `Start` is called (but errors are returned via `errChan`)

Reply:
i expect `New` to immediately fail on empty key/cert filepath, looks like it doesn't with proposed changes. Not critical for now since app checks that on config parsing, but still not very good to me